Pocket.Container.Tests/PocketContainerTestFakesStrategyTests.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft. All rights reserved. 
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Net.Http;
+using FluentAssertions;
+using Pocket.Container.Test.Assembly;
+using Xunit;
+
+namespace Pocket.Container.Tests
+{
+    public class PocketContainerTestFakesStrategyTests
+    {
+        [Fact]
+        public void When_looking_for_fakes_near_to_a_type_nested_class_is_preferred()
+        {
+            var container = new PocketContainer().ResolveWithFakeTypesCloseTo(this);
+
+            var resolved = container.Resolve<RealObject>();
```

Comment:
In actual usage, this might not give the desired result. Assuming that in the production code path, `RealObject` has been registered, then the strategy won't do anything. A strategy is only evaluated if there isn't a registration for the requested type. So I think the call to `ResolveWithFakeTypesCloseTo` would need to be in the production code, not the test code, meaning the word "fake" wouldn't apply. The strategy would really be more like `ResolveWithClosestType` or something like that.

Reply:
yes, I agree, will change this to ResolveWithClosestType strategy. I know that f the type is registered this won't be invoked but the goal of the code here is to be a strategy

Reply:
It might be clearer to use the word "use" rather than "resolve" since the various `Resolve*` methods are for actually creating instances, not for configuring the container.